doodad/mode.py

```diff
@@ -909,8 +909,15 @@ def create_instance(self, metadata, verbose=False):
             if isinstance(e, AzureCloudError):
                 print("Error when creating VM. Error message:")
                 print(e.message + '\n')
-
-                return success, e
+                if attempt_number <= 4:
+                    print("Retrying (attempt {})".format(attempt_number+1))
+                    return self.create_instance(
+                        metadata,
```

Comment:
Is retrying with the same metadata helpful? Maybe we could try a different region in the next attempt, something like: https://github.com/rail-berkeley/doodad/blob/2195b70b7405bebfe62e57ad8fac9d44ab49e8f7/doodad/mode.py#L683

Reply:
oh, that's a really nice idea! I did find that sometimes I just had to re-launch with the exact same metadata, and that helped

Reply:
Well, maybe rotating over region will make the doodad launch a bit more straightforward with Azure! If you would like integrate the  iteration over regions, I would recommend sticking to these here: https://github.com/rail-berkeley/doodad/blob/2195b70b7405bebfe62e57ad8fac9d44ab49e8f7/doodad/mode.py#L671 (lines 671, 672) since these have the most quota available.